### PR TITLE
V0.12 Patch increase gas limit for service account txs

### PR DESCRIPTION
**Pull request by ramtinms**
This PR 
  - allows any limit for transactions with service account as payer 
  - increase the collection total gas limit to 10M from 1M

### CODE REVIEW

File: cmd/collection/main.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func main() {
 				"maximum number of transactions in proposed collections")
 			flags.Uint64Var(&maxCollectionByteSize, "builder-max-collection-byte-size", 1000000,
 				"maximum byte size of the proposed collection")
-			flags.Uint64Var(&maxCollectionTotalGas, "builder-max-collection-total-gas", 1000000,
+			flags.Uint64Var(&maxCollectionTotalGas, "builder-max-collection-total-gas", 10000000,
```
Comment (Kay-Zee) [inline cmd/collection/main.go:103]: 1 million was probably fine, but i think 10million should still be okay?

File: access/validator.go [hunk 2]
```diff
@@ -141,6 +143,12 @@ func (v *TransactionValidator) checkMissingFields(tx *flow.TransactionBody) erro
 }
 
 func (v *TransactionValidator) checkGasLimit(tx *flow.TransactionBody) error {
+	// if service account is the payer of the transaction accepts any gas limit
+	// note that even though we don't enforce any limit here, exec node later
+	// enforce a max value for any transaction
+	if tx.Payer == v.serviceAccountAddress {
```
Comment (Kay-Zee) [inline access/validator.go:149]: any reason we didn't just check against `chain.ServiceAddress()` here? I'm guessing saving that function call?

**Review by Kay-Zee — APPROVED** [review]

File: access/validator.go (hunk 2 above)
Comment (ramtinms) [reply access/validator.go:149]: yeah not a cheap function to be called for every transaction, it creates a new address generator and asks for the first address.

File: cmd/collection/main.go (hunk 1 above)
Comment (ramtinms) [reply cmd/collection/main.go:103]: yeah better to go with the bigger number than the smaller one for now

**Review by jordanschalm — APPROVED** [review]

**Pull request merged**